isisd/isis_sr.c

```diff
@@ -674,12 +694,13 @@ static mpls_label_t sr_prefix_in_label(const struct sr_prefix *srp)
  */
 static mpls_label_t sr_prefix_out_label(const struct sr_prefix *srp,
 					const struct sr_node *srn_nexthop,
+					struct isis_route_info *rinfo,
 					const uint8_t *sysid)
 {
-	const struct sr_node *srn = srp->srn;
+	bool last_hop = (rinfo->depth == 2);
 
 	/* Check if the nexthop SR Node is the last hop? */
-	if (memcmp(sysid, srn->sysid, ISIS_SYS_ID_LEN) == 0) {
+	if (last_hop) {
 		/* SR-Node doesn't request NO-PHP. Return Implicit NULL label */
 		if (!CHECK_FLAG(srp->sid.flags, ISIS_PREFIX_SID_NO_PHP))
 			return MPLS_LABEL_IMPLICIT_NULL;
```

Comment:
maybe the change is no necessary ,
the sysid in nexthop means the node which have same sysid is connected . 
if the are different , the srn is no direct.
just my idea.